src/ipc-handlers.ts

```diff
@@ -479,12 +479,22 @@ export async function getSnapshots( _event: IpcMainInvokeEvent ): Promise< Snaps
 	return snapshots;
 }
 
-export async function openSiteURL( event: IpcMainInvokeEvent, id: string, relativeURL = '' ) {
+export async function openSiteURL(
+	event: IpcMainInvokeEvent,
+	id: string,
+	relativeURL = '',
+	{ autoLogin = true }: { autoLogin?: boolean } = {}
+) {
 	const site = SiteServer.get( id );
 	if ( ! site ) {
 		throw new Error( 'Site not found.' );
 	}
-	shell.openExternal( site.server?.url + relativeURL );
+	const url = new URL( site.server?.url + relativeURL );
```

Comment:
What do you think about checking the server has URL first? Something like:
```
	if ( ! site.server?.url ) {
		throw new Error( 'Site server URL not found.' );
	}
```

My concern is that site.server?.url could be undefined and the `new URL` would throw a "Invalid URL" exception when trying to create `undefined/wp-admin`.

Reply:
I've addressed this https://github.com/Automattic/studio/pull/532/commits/38d1d488300f002572943886c4ffa18f16e92609.